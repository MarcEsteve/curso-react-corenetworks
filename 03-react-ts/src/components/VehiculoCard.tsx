type Props = {
  matricula: string;
  modelo: string;
  color: string;
  itv: boolean;
};

export default function VehiculoCard({ matricula, modelo, color, itv }: Props) {
  return (
    <article>
      <h3>ðŸš— {modelo}</h3>
      <p>MatrÃ­cula: {matricula}</p>
      <p>Color: {color}</p>
      <p>ITV: {itv ? "Vigente" : "Caducada"}</p>
    </article>
  );
}
