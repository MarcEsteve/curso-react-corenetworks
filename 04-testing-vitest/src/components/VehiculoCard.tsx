type Props = { matricula: string; modelo: string; itv: boolean };

export default function VehiculoCard({ matricula, modelo, itv }: Props) {
  return (
    <article>
      <h3>ðŸš— {modelo}</h3>
      <p>MatrÃ­cula: {matricula}</p>
      <p>ITV: {itv ? "Vigente âœ…" : "Caducada ðŸš«"}</p>
    </article>
  );
}
