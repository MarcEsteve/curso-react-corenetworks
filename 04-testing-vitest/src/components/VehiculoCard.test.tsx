import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import VehiculoCard from "./VehiculoCard";

describe("VehiculoCard", () => {
  it("renderiza correctamente el componente (snapshot)", () => {
    const { container } = render(
      <VehiculoCard matricula="1234ABC" modelo="Toyota" itv={true} />
    );

    expect(container).toMatchSnapshot();
  });
});
