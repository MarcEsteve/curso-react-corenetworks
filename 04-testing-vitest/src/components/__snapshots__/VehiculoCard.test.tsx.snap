// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VehiculoCard > renderiza correctamente el componente (snapshot) 1`] = `
<div>
  <article>
    <h3>
      ðŸš— 
      Toyota
    </h3>
    <p>
      MatrÃ­cula: 
      1234ABC
    </p>
    <p>
      ITV: 
      Vigente âœ…
    </p>
  </article>
</div>
`;
